lib/IRGen/GenReflection.cpp

```diff
@@ -168,25 +168,29 @@ class PrintMetadataSource
   }
 };
 
-llvm::Constant *IRGenModule::getTypeRef(CanType type, MangledTypeRefRole role) {
+llvm::Constant *IRGenModule::getTypeRef(CanType type,
```

Comment:
May as well relax this to just Type so that callers don't have to call getCanonicalType()

Reply:
Yeah, I guess so.

Reply:
The whole point of taking a `CanType` here, and throughout IRGen, is that the type should already be correctly canonicalized by this point. Are we messing up earlier in the pipeline?